src/dialog/dlgreplacecuecolor.cpp

```diff
@@ -0,0 +1,319 @@
+                QStringLiteral(":current_color"));
+        queryValues.insert(QStringLiteral(":current_color"), mixxx::RgbColor::toQVariant(currentColor));
+    }
+    if (conditions.testFlag(ConditionFlag::HotcueIndexCheck)) {
+        queryStringConditions << QString(
+                QStringLiteral("hotcue") +
+                (conditions.testFlag(ConditionFlag::HotcueIndexNotEqual) ? QStringLiteral("!=") : QStringLiteral("=")) +
+                QStringLiteral(":hotcue"));
+        queryValues.insert(QStringLiteral(":hotcue"), QVariant(hotcueIndex));
+    }
+
+    QString queryString = QStringLiteral("SELECT id, track_id FROM " CUE_TABLE);
+    if (!queryStringConditions.isEmpty()) {
+        queryString += QStringLiteral(" WHERE ") + queryStringConditions.join(QStringLiteral(" AND "));
+    }
+
+    // Select affected queues
+    QSqlQuery query(database);
+    query.prepare(queryString);
+    for (auto i = queryValues.constBegin(); i != queryValues.constEnd(); i++) {
+        query.bindValue(i.key(), i.value());
+    }
+
+    if (!query.exec()) {
+        LOG_FAILED_QUERY(query);
+        return {};
+    }
+    QMap<int, int> ids;
+    int idColumn = query.record().indexOf("id");
+    int trackIdColumn = query.record().indexOf("track_id");
+    while (query.next()) {
+        ids.insert(query.value(idColumn).toInt(), query.value(trackIdColumn).toInt());
+    }
+
+    return ids;
+}
+
+void DlgReplaceCueColor::slotDatabaseIdsSelected() {
+    QMap<int, int> ids = m_dbSelectFuture.result();
+
+    if (ids.size() == 0) {
+        QMessageBox::warning(this, tr("No colors changed!"), tr("No cues matched the specified criteria."));
+        setApplyButtonEnabled(true);
+        return;
+    }
+
+    QSet<int> cueIds;
+    QSet<TrackId> trackIds;
+    for (auto i = ids.constBegin(); i != ids.constEnd(); i++) {
+        cueIds << i.key();
+        trackIds << TrackId(i.value());
+    }
+
+    if (QMessageBox::question(
+                this,
+                tr("Really replace colors?"),
+                tr("Really replace the colors of %1 cues in %2 tracks? This change cannot be undone!").arg(QString::number(cueIds.size()), QString::number(trackIds.size()))) == QMessageBox::No) {
+        setApplyButtonEnabled(true);
+        return;
+    }
+
+    mixxx::RgbColor::optional_t newColor = mixxx::RgbColor::fromQString(pushButtonNewColor->text());
+    VERIFY_OR_DEBUG_ASSERT(newColor) {
+        setApplyButtonEnabled(true);
+        return;
+    }
+
+    m_dbUpdateFuture = QtConcurrent::run(
+            this,
+            &DlgReplaceCueColor::updateCues,
+            cueIds,
+            trackIds,
+            *newColor);
+    m_dbUpdateFutureWatcher.setFuture(m_dbUpdateFuture);
+}
+
+void DlgReplaceCueColor::updateCues(QSet<int> cueIds, QSet<TrackId> trackIds, mixxx::RgbColor newColor) {
+    // The pooler limits the lifetime all thread-local connections,
+    // that should be closed immediately before exiting this function.
```

Comment:
do you mean it will be closed by the destructor when it goes out of scope, or the programmer must remember to close it?  If it's the former, I think this comment is unnecessary because you're just describing the API.

Reply:
The former, but it explains why in create an instance of the pooler but never do something with it. Should I remove the comment anyway?

Reply:
You could do it for the first instance of this idiom in the file, if you want.  But this idiom (RAII) is a fundamental pattern in C++ so it should be understood.  However since we have a lot of beginners on mixxx it might help.  But please reword a bit:

"The pooler limits the lifetime of all thread-local connections.  It will be closed by its destructor when it goes out of scope"